BLE_GattServer_FOTAService/MCUboot/client/client.py

```diff
@@ -0,0 +1,330 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+import hashlib
+import logging
+import asyncio
+
+from bleak          import BleakScanner, BleakClient, BleakError
+from bleak.uuids    import uuid16_dict
+from typing         import Optional, Union
+from logging.config import fileConfig
+
+uuid16_dict = {v: k for k, v in uuid16_dict.items()}
+
+fileConfig('logging.conf')
+
+logger = logging.getLogger('logger')
+
+# FOTA UUIDS
+UUID_FOTA_SERVICE                  = "53880000-65fd-4651-ba8e-91527f06c887"
+UUID_BINARY_STREAM_CHAR            = "53880001-65fd-4651-ba8e-91527f06c887"
+UUID_CONTROL_CHAR                  = "53880002-65fd-4651-ba8e-91527f06c887"
+UUID_STATUS_CHAR                   = "53880003-65fd-4651-ba8e-91527f06c887"
+UUID_VERSION_CHAR                  = "53880004-65fd-4651-ba8e-91527f06c887"
+UUID_FIRMWARE_REVISION_STRING_CHAR = "0000{0:x}-0000-1000-8000-00805f9b34fb".format(
+    uuid16_dict.get("Firmware Revision String")
+)
+
+# Additional UUIDs
+UUID_DEVICE_INFORMATION_SERVICE    = "0000{0:x}-0000-1000-8000-00805f9b34fb".format(
+    uuid16_dict.get("Device Information")
+)
+UUID_CUDD                          = "0000{0:x}-0000-1000-8000-00805f9b34fb".format(
+    uuid16_dict.get("Characteristic User Description")
+)
+
+# FOTA status codes
+FOTA_STATUS_OK                     = bytearray(b'\x00')
+FOTA_STATUS_UPDATE_SUCCESSFUL      = bytearray(b'\x01')
+FOTA_STATUS_XOFF                   = bytearray(b'\x02')
+FOTA_STATUS_XON                    = bytearray(b'\x03')
+FOTA_STATUS_SYNC_LOST              = bytearray(b'\x04')
+FOTA_STATUS_UNSPECIFIED_ERROR      = bytearray(b'\x05')
+FOTA_STATUS_VALIDATION_FAILURE     = bytearray(b'\x06')
+FOTA_STATUS_INSTALLATION_FAILURE   = bytearray(b'\x07')
+FOTA_STATUS_OUT_OF_MEMORY          = bytearray(b'\x08')
+FOTA_STATUS_MEMORY_ERROR           = bytearray(b'\x09')
+FOTA_STATUS_HARDWARE_ERROR         = bytearray(b'\x0a')
+FOTA_STATUS_NO_FOTA_SESSION        = bytearray(b'\x0b')
+
+# FOTA op codes
+FOTA_OP_CODE_START                 = bytearray(b'\x01')
+FOTA_OP_CODE_STOP                  = bytearray(b'\x02')
+FOTA_OP_CODE_COMMIT                = bytearray(b'\x03')
+FOTA_OP_CODE_SET_XOFF              = bytearray(b'\x41')
+FOTA_OP_CODE_SET_XON               = bytearray(b'\x42')
+FOTA_OP_CODE_SET_FRAGMENT_ID       = bytearray(b'\x43')
+
+# Number of bytes in a single BSC packet, excluding fragment ID
+FRAGMENT_SIZE       = 128
+
+# Max. Frag. ID before overflow to zero
+MAXIMUM_FRAGMENT_ID = 255
+
+
+def get_chunk_n(data, chunk_size: int, n: int):
+    start = chunk_size * n
+    end = chunk_size * (n+1)
+    if len(data) <= start:
+        return []
+    if len(data) <= end:
+        return data[start:]
+
+    return data[start:end]
+
+
+async def find_device(name: str) -> Optional[BleakClient]:
```

Comment:
Since the function connects to the device, maybe it would be more appropriate to rename it `connect`. What do you think ?

Reply:
Sure. I'll rename it.